### PR TITLE
Decouple more json parsing logic

### PR DESCRIPTION
**Pull request by felipecsl**
This was a bit more involved as the JSON parsing in  `Keyframe` and `AnimatableValue` classes is a lot more involved.
All tests still pass but this change makes me a bit more nervous, @gpeal 
Please take a closer look if you can.

Part of #3

### CODE REVIEW

File: lottie/src/main/java/com/airbnb/lottie/Keyframe.java [hunk 1]
```diff
@@ -52,62 +38,22 @@ static void setEndFrames(List<? extends Keyframe<?>> keyframes) {
...
 
   private final LottieComposition composition;
-  T startValue;
-  T endValue;
-  Interpolator interpolator;
-  @SuppressWarnings("WeakerAccess") float startFrame;
-  @SuppressWarnings("WeakerAccess") @Nullable Float endFrame = null;
-
-  public Keyframe(LottieComposition composition, float startFrame, float endFrame) {
+  @Nullable T startValue;
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/Keyframe.java]: Can these be final?

File: lottie/src/main/java/com/airbnb/lottie/Layer.java [hunk 2]
```diff
@@ -244,30 +244,24 @@ static Layer newInstance(JSONObject json, LottieComposition composition) {
         preCompHeight = (int) (json.optInt("h") * composition.getScale());
       }
 
-      long inFrame = json.optLong("ip");
-      long outFrame = json.optLong("op");
+      float inFrame = json.optLong("ip");
```
Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/Layer.java]: Hmm, I don't think these can be float values. It's getting a long from the json anyways.
Can you change the json to `optInt` and all of the vars to int and Integer?

File: lottie/src/main/java/com/airbnb/lottie/ColorFactory.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.airbnb.lottie;
+
+import android.graphics.Color;
+
+import org.json.JSONArray;
+
+class ColorFactory {
```
Comment (felipecsl) [inline lottie/src/main/java/com/airbnb/lottie/ColorFactory.java]: sure

File: lottie/src/main/java/com/airbnb/lottie/Keyframe.java (hunk 1 above)
Comment (felipecsl) [inline lottie/src/main/java/com/airbnb/lottie/Keyframe.java]: They are modified from `AnimatableIntegerValue`

File: lottie/src/main/java/com/airbnb/lottie/Layer.java (hunk 2 above)
Comment (felipecsl) [inline lottie/src/main/java/com/airbnb/lottie/Layer.java]: This means that `inOutKeyframes` needs to change to `List<Keyframe<Integer>>`. Is that correct?

Comment (gpeal) [inline lottie/src/main/java/com/airbnb/lottie/Layer.java]: @felipecsl yeah that's correct.

**Pull request merged**